### PR TITLE
[Tests-Only] add api test different encoding with encryption

### PR DESCRIPTION
**Pull request by dpakach**
add tests for different encoding config with encryption

### CODE REVIEW

**Comment by dpakach** [comment]
I made the PR to master because making PR to the branch `binary-encoding` doesn't starts the CI, we can change the base to that branch later.

**Comment by karakayasemi** [comment]
@dpakach after this PR approved, I can cherry-pick your commits into top of the `binary-encode` branch.

**Comment by phil-davis** [comment]
@dpakach #237 has been merged. You can rebase this and we can move forward...

File: tests/acceptance/features/apiEncryption/encoding.feature [hunk 1]
```diff
@@ -0,0 +1,81 @@
+@api
+Feature: encrypt files with legacy and default(new) encoding
+
+  Background:
+    Given user "Alice" has been created with default attributes and without skeleton files
+    And user "Alice" has uploaded file with content "some data" to "/textfile0.txt"
```
Comment (phil-davis) [inline tests/acceptance/features/apiEncryption/encoding.feature:6]: why do we need this step?
I don't see where `textfile0.txt` is used.

File: tests/acceptance/features/apiEncryption/encoding.feature [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    And user "Brian" has been created with default attributes and without skeleton files
+    And user "Alice" has shared file "textfile1.txt" with user "Brian"
+    Then the content of file "/textfile1.txt" for user "Brian" should be "some data"
+
+  Scenario: share file with default encoding
+    Given the administrator has added system config key "encryption.use_legacy_encoding" with value "false" and type "boolean"
+    And user "Alice" has uploaded file with content "some data" to "/textfile1.txt"
+    And user "Brian" has been created with default attributes and without skeleton files
+    And user "Alice" has shared file "textfile1.txt" with user "Brian"
+    Then the content of file "/textfile1.txt" for user "Brian" should be "some data"
```
Comment (phil-davis) [inline tests/acceptance/features/apiEncryption/encoding.feature:51]: There could also be a scenario where the file is uploaded and shared when legacy encoding is in effect. Then switch to new encoding and check that Brian can still download the file. And a check that Brian can overwrite the file and Alice can see the changed content.

That should be enough - to do more we would end up writing scenarios for all the combinations of group shares, public link shares, with files in folders, files in sub-folders...!

File: tests/acceptance/features/apiEncryption/encoding.feature [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  Scenario: User overwrites a shared file with legacy encoding after encoding is switched
+    Given the administrator has added system config key "encryption.use_legacy_encoding" with value "false" and type "boolean"
+    And user "Alice" has uploaded file with content "some data" to "/textfile1.txt"
+    And user "Brian" has been created with default attributes and without skeleton files
+    And user "Alice" has shared file "textfile1.txt" with user "Brian"
+    And the administrator has added system config key "encryption.use_legacy_encoding" with value "true" and type "boolean"
+    Then the content of file "/textfile1.txt" for user "Brian" should be "some data"
+    When user "Brian" uploads file with content "some new data" to "/textfile1.txt" using the WebDAV API
+    Then the content of file "/textfile1.txt" for user "Alice" should be "some new data"
+    And the content of file "/textfile1.txt" for user "Alice" should be "some new data"
```
Comment (phil-davis) [inline tests/acceptance/features/apiEncryption/encoding.feature:91]:
```suggestion
    And the content of file "/textfile1.txt" for user "Brian" should be "some new data"
```

File: tests/acceptance/features/apiEncryption/encoding.feature [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  Scenario: User overwrites a shared file with default encoding after encoding is switched
+    Given the administrator has added system config key "encryption.use_legacy_encoding" with value "true" and type "boolean"
+    And user "Alice" has uploaded file with content "some data" to "/textfile1.txt"
+    And user "Brian" has been created with default attributes and without skeleton files
+    And user "Alice" has shared file "textfile1.txt" with user "Brian"
+    And the administrator has added system config key "encryption.use_legacy_encoding" with value "false" and type "boolean"
+    Then the content of file "/textfile1.txt" for user "Brian" should be "some data"
+    When user "Brian" uploads file with content "some new data" to "/textfile1.txt" using the WebDAV API
+    Then the content of file "/textfile1.txt" for user "Alice" should be "some new data"
+    And the content of file "/textfile1.txt" for user "Alice" should be "some new data"
```
Comment (phil-davis) [inline tests/acceptance/features/apiEncryption/encoding.feature:102]:
```suggestion
    And the content of file "/textfile1.txt" for user "Brian" should be "some new data"
```

**Review by phil-davis — COMMENTED** [review]
Looks good now. @dpakach ~you can cherry-pick the commit with tests onto the other PR~ - I have done it

**Comment by dpakach** [comment]
closing this since tests have been moved to https://github.com/owncloud/encryption/pull/224/commits

**Pull request closed without merging**